tsl/src/continuous_aggs/create.c

```diff
@@ -2064,11 +2088,12 @@ build_union_query(CAggTimebucketInfo *tbinfo, MatTableColumnInfo *mattblinfo, Qu
 	attno =
 		get_attnum(tbinfo->htoid, get_attname_compat(tbinfo->htoid, tbinfo->htpartcolno, false));
 	varno = list_length(q2->rtable);
-	q2->jointree->quals = build_union_query_quals(tbinfo->htid,
-												  tbinfo->htpartcoltype,
-												  get_negator(tce->lt_opr),
-												  varno,
-												  attno);
+	q2_quals = build_union_query_quals(tbinfo->htid,
+									   tbinfo->htpartcoltype,
+									   get_negator(tce->lt_opr),
+									   varno,
+									   attno);
+	q2->jointree->quals = make_and_qual(q2->jointree->quals, q2_quals);
```

Comment:
quals are implicitly ANDed, can you elaborate why you introduce nesting here?

Reply:
Made the and explicit. Is there a reason not to do it this way?

Reply:
You introduce nesting lets say we have 3 quals q1, q2, q3 the usual representation would be in a flat list AND(q1, q2, q3) you are storing them AS AND(q1, AND(q2, q3))

Reply:
jointree->quals are sometimes not Lists. (e.g.  col IN (1,2) ).